src/lib/y2partitioner/widgets/delete_disk_partition_button.rb

```diff
@@ -48,7 +49,37 @@ def handle
 
     private
 
+      def confirm(device)
+        names = children_names(device)
+
+        if names.empty?
+          Yast::Popup.YesNo(
+            # TRANSLATORS %s is device to be deleted
+            format(_("Really delete %s?"), device.name)
+          )
+        else
+          confirm_recursive_delete(
+            names,
+            _("Confirm Deleting of All Partitions"),
+            # TRANSLATORS: type stands for type of device and name is its identifier
+            format(_("The %{type} \"%{name}\" contains at least one another device.\n" \
+              "If you proceed, the following devices will be deleted:"),
+              name: device.name,
+              type: device.is?(:disk) ? _("disk") : _("partition")),
+            format(_("Really delete all devices on \"%s\"?"), device.name)
+          )
+        end
+      end
+
+      def children_names(device)
+        device.descendants.map do |dev|
+          dev.name if dev.respond_to?(:name)
+        end.compact
```

Comment:
I'm fine with this. But somebody I know called J.R. would ask you to use `each_with_object` here.